### PR TITLE
Fix error: "Can't find variable: TextDecoder" 

### PR DESCRIPTION
**Pull request by keithwrightbos**
Occurs on browsers that do no support TextDecoder/TextEncoder (e.g. safari) due to check no properly prefacing with self (aka window)

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -764,7 +764,8 @@ export class AmpA4A extends AMP.BaseElement {
               return true;
             });
         } catch (e) {
-          dev().error('Error injecting creative in friendly frame', e);
+          dev().error(
+            'Amp Ad', 'Error injecting creative in friendly frame', e);
```
Comment (jridgewell) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Undo this one, I've already got a [fix](https://github.com/ampproject/amphtml/pull/6005) for it.

File: src/utils/bytes.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ import {dev} from '../log';
  * @return {!Promise<string>}
  */
 export function utf8Decode(bytes) {
-  if (TextDecoder) {
+  if (self.TextDecoder) {
```
Comment (jridgewell) [inline src/utils/bytes.js]: Preference on this vs `typeof X === 'undefined'`? The later's the way I usually see this.

**Comment by aghassemi** [comment]
@keithwrightbos @tdrl Does this need to be cherry picked for Canary and/or Prod?

**Comment by keithwrightbos** [comment]
Ideally yes.  Give me an hour and I will address Justin's feedback.  Will
update PR when done

**Comment by keithwrightbos** [comment]
@jridgewell should be ready to merge.  Please let me know when this has been patched and deployed to production.  Thanks

File: src/utils/bytes.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ import {dev} from '../log';
  * @return {!Promise<string>}
  */
 export function utf8Decode(bytes) {
-  if (TextDecoder) {
+  if (typeof self.TextDecoder !== 'undefined') {
```
Comment (jridgewell) [inline src/utils/bytes.js]: Remove the `self.`, it won't be necessary with `typeof`.

File: src/utils/bytes.js [hunk 4]
```diff
@@ -44,7 +44,7 @@ export function utf8Decode(bytes) {
  * @return {!Promise<!Uint8Array>}
  */
 export function utf8Encode(string) {
-  if (TextEncoder) {
+  if (typeof self.TextEncoder !== 'undefined') {
```
Comment (jridgewell) [inline src/utils/bytes.js]: Ditto.

**Pull request merged**